### ISSUE TITLE
Line breaks in RTL languages (Arabic in this case)

### ISSUE THREAD

**Issue by GregSharp-app** [pdfme/pdfme#398]
### Describe the bug

Hi everyone,
Having a long string (with no line breaks)  results in weird line breaks :
![image](https://github.com/pdfme/pdfme/assets/73749174/e32c3b53-6bc4-4fce-8af6-099bdfc76160)

this is the design template 
![image](https://github.com/pdfme/pdfme/assets/73749174/f4c03ebb-a1fb-4e06-9bc0-f8f2539d078a)

And these are the parameters
![image](https://github.com/pdfme/pdfme/assets/73749174/cad28793-f0d7-483e-bc50-bea98308d593)

i tried to changes petameters (alignment, font size ..) but it still resulted in those line breaks. I temporarily fixed the issue by counting how many Arabic characters can fit in that rectangle without line break, and then i manually add line breaks :
`
function getDescription(text?: string): string {
	const words = (text?.split(/\s+/) ?? []).filter((c) => !['\n', '\r'].includes(c));
        // number of Arabic characters (approximately) that can fit without resulting in line break 
	const maxCharacters = 80;
	let charCounter = 0;
	let result = '';

	for (const word of words) {
		charCounter += word.length;

		if (charCounter <= maxCharacters) {
			result += word + ' ';
		} else {
			result += '\n' + word + ' ';
			charCounter = word.length;
		}
	}

	return result.trim();
}
`
caveman solution but for now it seems to work
![image](https://github.com/pdfme/pdfme/assets/73749174/99afe1f4-88c8-46a7-bd74-894fc2044518)

I was wondering if there's a solution for this, or am i just doing something wrong.
Have a great day everyone 🙏

### To Reproduce

Just use a font that support arabic characters and a long arabic text with no line breaks

### Expected behavior

for line breaks to properly work 

### Your Environment

```markdown
- pdfme package(@pdfme/generator or @pdfme/ui):
- pdfme version:
- Operating system:
- Node.js version or Browser name & version:
```

### Your Error Log

```shell
-
```

**Comment by peteward** [comment]
Hi @GregSharp-app,

Thanks for reporting this bug.

Could you share both the font you are using and the text that's causing the issue? I can take a look and see what's going on.

**Comment by GregSharp-app** [comment]
Hi @peteward, i hope you're doing well,
the font i used is 'Neo Sans Arabic' i also tried with 'Neo Sans Arabic' and 'Cairo' but the bug still  occurred,
https://www.cufonfonts.com/font/neo-sans-arabic
https://fonts.google.com/specimen/Cairo
https://fonts.google.com/noto/specimen/Noto+Sans+Arabic
------
the text is: 
`يهدف المشروع إلى إنشاء متجر تجزئة متخصص في بيع مجموعة واسعة من منتجات ومعدات الطاقة الشمسية. سيوفر هذا المتجر مجموعة واسعة من الألواح الشمسية، وأضواء تعمل بالطاقة الشمسية، وشواحن، وبطاريات، وملحقات أخرى ذات صلة. سيكون التركيز على توفير تقنية الطاقة الشمسية عالية الجودة والموثوقة لتلبية الطلب المتزايد على الحلول البيئية للطاقة. بالإضافة إلى ذلك، سيقدم المتجر الإرشادات والمعلومات الخبيرة للعملاء حول اختيار أفضل منتجات الطاقة الشمسية المناسبةيهدف المشروع إلى إنشاء متجر تجزئة متخصص في بيع مجموعة واسعة من منتجات ومعدات الطاقة الشمسية.`

Here is the template i used, i deleted other field names except 'projectDescription',
[Template Description Bug.json](https://github.com/pdfme/pdfme/files/13823296/Template.Description.Bug.json)

Best regards.

**Comment by peteward** [comment]
Hi @GregSharp-app,

Many thanks for sharing. I will try to take a look at the weekend.

**Comment by GregSharp-app** [comment]
Sure 👍
If there's anything i can help with ,please do tell

**Comment by peteward** [comment]
Hi @GregSharp-app ,

Apologies for the slow follow-up, I had a bad illness for a few weeks but am back now and catching up.

I have managed to reproduce this locally. 

The pdfme library calculates where lines should be split based on the width of the containing box. We do this for dynamic font size calculation, text alignment and character spacing.

pdflib, which we use for rendering the pdfs also has the ability to line split, but we override this to provide the functionality above. The problem here is that pdflib thinks the lines that we have already calculated and split are too long and wraps them again, overlaying them on top of the lines that we are drawing.  We saw this before when we had bugs in the line splitting logic but this was resolved.

I'm fairly sure that our (pdfme) calculation for splitting lines is correct, even for these Arabic fonts, is correct and the problem lies in pdflib. If I pass a much larger max-width to the pdf rendering function then it appears to render correctly for the few tests that I've tried.

I've run out of time now but will continue the investigation soon.
